lib/rspec/core/notifications.rb

```diff
@@ -199,6 +199,12 @@ def fully_formatted(failure_number, colorizer=::RSpec::Core::Formatters::Console
         @exception_presenter.fully_formatted(failure_number, colorizer)
       end
 
+      # @return [Array] The failure information fully formatted in the way that
+      #   RSpec's built-in formatters emit, split by line.
+      def fully_formatted_lines(failure_number, colorizer=::RSpec::Core::Formatters::ConsoleCodes)
+        @exception_presenter.fully_formatted_lines(failure_number, colorizer)
+      end
```

Comment:
As above that would actually make it problematic for removing the first two bits of the message, and given that we've needed this for our internal formatters, its likely that it would be useful to other people, I think thats justification enough for the minimal extra burden.

It's currently quite difficult to modify the how these messages are generated internally, and as you've rightly pointed out regular expressions to modify the output can be unweildy and difficult to grok, adding a minimal extra method to remove some of that seems worth it to me, your mileage may vary of course :)